src/software/sensor_fusion/filter/ball_filter.h

```diff
@@ -80,65 +115,90 @@ class BallFilter
+     * @throws std::invalid_argument if ball_detections has less than 2 elements
+     *
+     * @param ball_detections The ball detections to fit
+     *
+     * @return The line of best fit through the given ball detection positions
      */
-    std::optional<BallVelocityEstimate> estimateBallVelocity(
-        boost::circular_buffer<BallDetection> ball_detections,
-        const std::optional<Line> &ball_regression_line = std::nullopt);
+    static Line calculateLineOfBestFit(
+        boost::circular_buffer<BallDetection> ball_detections);
 
     /**
      * Given a list of ball detections, use linear regression to find a line of best fit
      * through the ball positions, and calculate the error of this regression.
      *
+     * @pre ball_detections must have at least 2 elements
+     *
+     * @throws std::invalid_argument if ball_detections has less than 2 elements
+     *
      * @param ball_detections The ball detections to use in the regression
+     *
      * @return A struct containing the regression line and error of the linear regression
      */
-    LinearRegressionResults getLinearRegressionLine(
+    static LinearRegressionResults calculateLinearRegression(
         boost::circular_buffer<BallDetection> ball_detections);
 
     /**
-     * Uses linear regression to filter the given list of ball detections to fine the
-     * current "real" state of the ball.
+     * Returns the filtered position of the ball given a buffer of ball detections
+     * and the line of best fit through them.
      *
-     * @param ball_detections The detections to filter
-     * @return The filtered current state of the ball. If a filtered result cannot be
-     * calculated, returns std::nullopt
+     * @pre ball_detections must be non-empty
+     *
+     * @throws std::invalid_argument if ball_detections has less than 2 elements
+     *
+     * @param ball_detections The ball detections
+     * @param regression_line The line of best fit through the ball positions
+     *
+     * @return The filtered position of the ball
      */
-    std::optional<TimestampedBallState> estimateBallState(
-        boost::circular_buffer<BallDetection> ball_detections);
+    static Point getFilteredPosition(
+        boost::circular_buffer<BallDetection> ball_detections,
+        const Line& regression_line);
 
-   private:
-    unsigned int _min_buffer_size;
-    unsigned int _max_buffer_size;
+    /**
+     * Returns the filtered velocity of the ball given a buffer of ball detections
+     * and the line of best fit through them.
+     *
+     * @param ball_detections The ball detections
+     * @param regression_line The line of best fit through the ball positions
+     *
+     * @return The filtered velocity of the ball. If an error occurs that prevents
+     * the velocity from being calculated properly, std::nullopt is returned.
+     */
+    static std::optional<Vector> getFilteredVelocity(
```

Comment:
What's the difference between this and estimateBallVelocity? I think the naming is a bit confusing

Reply:
Now that you mention it, very little. I'll try combine them.

Reply:
done